equinox/_module.py

```diff
@@ -680,6 +680,15 @@ def __init__(self, method):
         self.method = method
         if getattr(self.method, "__isabstractmethod__", False):
             self.__isabstractmethod__ = self.method.__isabstractmethod__
+        # Fixes https://github.com/patrick-kidger/equinox/issues/1016
+        try:
+            self.__module__ = self.method.__module__
+        except AttributeError:
+            pass
+        try:
+            self.__doc__ = self.method.__doc__
+        except AttributeError:
+            pass
```

Comment:
Could each of these be `@property`s instead?

Reply:
I guess it should, see below, you prefer this way?

```python
>>> class Foo:
...   def __init__(self):
...     self.bar = 1
... 
>>> class Baz:
...   @property
...   def bar(self):
...     return 1
... 
>>> hasattr(Foo(), "bar")
True
>>> hasattr(Baz(), "bar")
True
>>> 
```